### ISSUE TITLE
Data stops coming

### ISSUE THREAD

**Issue by dortanes** [Streampunk/grandiose#30]
I discovered another problem: if the connection is bad, data stops coming (data(), video(), audio()).
Promise in data() runs endlessly. video() and audio() is going to timeout and isn't returning any data.

**Comment by ZaynSiv09** [comment]
That's true, so I hope to terminate these objects through a "destory()" method and let them be GC, which has been bothering me for a long time

**Comment by ZaynSiv09** [comment]
> I discovered another problem: if the connection is bad, data stops coming (data(), video(), audio()). Promise in data() runs endlessly. video() and audio() is going to timeout and isn't returning any data.

That's true, so I hope to terminate these objects through a "destory()" method and let them be GC, which has been bothering me for a long time
